### PR TITLE
relax rel param assertions for second infection

### PR DESCRIPTION
**Pull request by RaphaelS1**
* Allows relative parameters to be <1 for second infections
* Removes from Rt calculation:  mean_duration_G_D/hosp/icu (and associated tests)

### CODE REVIEW

File: R/carehomes.R [hunk 1]
```diff
@@ -112,8 +112,8 @@ NULL
 ##'   different relative susceptibilities by age (first dimension of the array),
 ##'   pathogen strain (second dimension) and vaccination group
 ##'   (third dimension); in that case, the first layer (3rd dimension) of
-##'   rel_susceptibility should be 1 (for the non-vaccinated group) and other
-##'   values be between 0 and 1
+##'   rel_susceptibility should be 1 (for the non-vaccinated group) for the
+##'   first two columns (first infection) and other values between 0 and 1
```
Comment (annecori) [inline R/carehomes.R:116]: "first infection" --> "first infection with either strain"?

File: R/carehomes.R [hunk 2]
```diff
@@ -128,7 +128,8 @@ NULL
 ##'   dimension of the array), pathogen strain (second dimension) and
 ##'   vaccination group (third dimension); in that case,
 ##'   the first layer of rel_p_sympt should be 1 (for the non-vaccinated group)
-##'   and other values be between 0 and 1
+##'   for the first two columns (first infection) and other values between 0
```
Comment (annecori) [inline R/carehomes.R:131]: "first infection" --> "first infection with either strain"?

File: R/carehomes.R [hunk 3]
```diff
@@ -143,7 +144,8 @@ NULL
 ##'   (first dimension of the array), pathogen strain (second dimension) and
 ##'   vaccination group (third dimension); in that case,
 ##'   the first layer of rel_p_hosp_if_sympt should be 1 (for the
-##'   non-vaccinated group) and other values be between 0 and 1
+##'   non-vaccinated group) for the first two columns (first infection) and
```
Comment (annecori) [inline R/carehomes.R:147]: "first infection" --> "first infection with either strain"?

File: R/carehomes.R [hunk 4]
```diff
@@ -157,7 +159,8 @@ NULL
 ##'   (first dimension of the array), pathogen strain (second dimension) and
 ##'   vaccination group (third dimension); in that case,
 ##'   the first layer of rel_infectivity should be 1 (for the
-##'   non-vaccinated group) and other values be between 0 and 1
+##'   non-vaccinated group) for the first two columns (first infection) and
```
Comment (annecori) [inline R/carehomes.R:162]: "first infection" --> "first infection with either strain"?

File: R/vaccination.R [hunk 5]
```diff
@@ -89,8 +89,10 @@ check_rel_param <- function(rel_param, name_param) {
   if (length(rel_param) == 0) {
     stop(paste("At least one value required for", name_param))
   }
-  if (!all(rel_param[, , 1] == 1)) {
+  if (ncol(rel_param) == 1 && !all(rel_param[, , 1] == 1)) {
     stop(paste("First value of", name_param, "must be 1"))
+  } else if (ncol(rel_param) > 1 && !all(rel_param[, 1:2, 1] == 1)) {
+    stop(paste("First value of", name_param, "must be 1 for first infection"))
```
Comment (annecori) [inline R/vaccination.R:95]: "first infection" --> "first infection with either strain"?

File: NEWS.md [hunk 6]
```diff
@@ -1,7 +1,12 @@
-# sircovid 0.10.50
+# sircovid 0.10.51
 
 * Fix Rt calculations for multistrain models with cross-immunity
 
+# sircovid 0.10.50
+
+* rel vacc params can now be less than one in unvaccinated classes for second infections (strains 3 and 4)
+
+
```
Comment (edknock) [inline NEWS.md:9]: The news items are the wrong way round for 0.10.50 and 0.10.51

File: R/carehomes_rt.R [hunk 7]
```diff
@@ -63,6 +63,11 @@ carehomes_Rt <- function(step, S, p, prob_strain = NULL,
                          eigen_method = "power_iteration", R = NULL,
                          weight_Rt = FALSE) {
 
+  if (sum(p$hosp_transmission, p$ICU_transmission, p$G_D_transmission) > 0) {
+    stop("Cannot currently compute Rt if any of 'hosp_transmission',
+    'ICU_transmission', 'G_D_transmission")
```
Comment (edknock) [inline R/carehomes_rt.R:68]:
```suggestion
    'ICU_transmission' or 'G_D_transmission' are non-zero")
```

**Review by edknock — APPROVED** [review]

**Pull request merged**